pkg/operator/operator.go

```diff
@@ -127,7 +128,8 @@ func NewOperator(ctx context.Context, operator *operator.Operator) (context.Cont
 		ec2api,
 		*sess.Config.Region,
 	)
-	versionProvider := version.NewProvider(operator.KubernetesInterface, cache.New(awscache.DefaultTTL, awscache.DefaultCleanupInterval))
+	versionProvider := version.NewProvider(operator.KubernetesInterface, cache.New(awscache.DefaultTTL, awscache.DefaultCleanupInterval),
+		operator.GetClient(), lo.Must(getHTTPClient(operator.GetConfig())))
```

Comment:
Should this be part of building the operator? Like `operator.GetHTTPClient()`

Reply:
Since we only need it for the version provider, I think it's sufficient to just construct it inline and then pass it into the one provider. If there's other providers that need the HTTP client, I can imagine us pulling this out in the future.